crates/runtime/src/dataconnector/listing/connector.rs

```diff
@@ -387,6 +400,72 @@ impl<T: ListingTableConnector + Display> DataConnector for T {
     }
 }
 
+/// Lists the available files for a ListingTableConnector/ObjectStore
+/// Infers if the `file_format` specified is valid, based on the existence of files with the required extension
+///
+/// # Errors
+///
+/// - If no files are found at the specified path
+/// - If no files with the specified extension are found
+async fn check_for_files_and_extensions(
+    dataconnector: String,
+    extension: &str,
+    table_path: ListingTableUrl,
+    ctx: &SessionContext,
+    object_store: &Arc<dyn ObjectStore>,
+) -> DataConnectorResult<()> {
+    let files: Vec<_> = table_path
+        .list_all_files(&ctx.state(), object_store, "")
+        .await
+        .map_err(|err| DataConnectorError::UnableToConnectInternal {
+            dataconnector: dataconnector.clone(),
+            source: err.into(),
+        })?
+        .try_collect()
+        .await
+        .map_err(|err| DataConnectorError::UnableToConnectInternal {
+            dataconnector: dataconnector.clone(),
+            source: err.into(),
+        })?;
+
+    if files.is_empty() {
+        return Err(DataConnectorError::InvalidConfigurationNoSource {
+            dataconnector: dataconnector.clone(),
+            message:
+                // Url could contain access keys from e.g. S3, so we don't want to log it.
+                "Failed to find any files at the specified path. Check the path and try again."
+                    .to_string(),
+        });
+    }
+
+    let extensions = files
+        .iter()
+        .filter_map(|file| file.location.extension().map(|e| format!(".{e}")))
+        .collect::<HashSet<_>>();
+
+    if !extensions.contains(extension) {
+        if extensions.is_empty() {
+            return Err(DataConnectorError::InvalidConfigurationNoSource {
+                dataconnector: dataconnector.clone(),
+                message: format!("Failed to find any files matching the extension '{extension}'.\nSpice could not find any files with extensions at the specified path. Check the path and try again."),
+            });
+        }
+
+        let display_extensions = extensions
+            .iter()
+            .map(|e| format!("'{e}'"))
+            .collect::<Vec<_>>()
+            .join(", ");
+
+        return Err(DataConnectorError::InvalidConfigurationNoSource {
+            dataconnector: dataconnector.clone(),
+            message: format!("Failed to find any files matching the extension '{extension}'.\nIs your `file_format` parameter correct? Spice found the following file extensions: {display_extensions}.\nFor further information, visit: https://docs.spiceai.org/components/data-connectors#object-store-file-formats")
```

Comment:
nit: Could be worth being its own error type.

Reply:
Is there a non-object-store connector that could use this? This error applies to every object store connector, which includes `file`, `s3`, `abfs` and `ftp/sftp`. I'm not sure there's another connector that would use this.